### PR TITLE
Added support for IMap.computeIfAbsent() #14913

### PR DESCRIPTION
**Pull request by webashutosh**
Added support for IMap.computeIfAbsent()

### CODE REVIEW

**Comment by webashutosh** [comment]
Hi @mmedenjak @Holmistr @sancar - this one is for `computeIfAbsent()` - could you please have a look?
I am planning to raise a PR each time I address a set of similar methods. 
computeIf* methods will be done with this PR.

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java [hunk 1]
```diff
@@ -2014,4 +2015,32 @@ private V computeIfPresentLocally(K key,
...
+
+    @Override
+    public V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(mappingFunction, NULL_FUNCTION_IS_NOT_ALLOWED);
+
+        return computeIfAbsentLocally(key, mappingFunction);
+    }
+
+    private V computeIfAbsentLocally(K key, Function<? super K, ? extends V> mappingFunction) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java]: Should the actual logic not be done on the server side?

It seems you are first getting the value, then convert it and try to set it. 

If it is all done remotely, less remote calls are involved.

Comment (webashutosh) [reply hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java]: Hi Peter,
For the client, it was decided to execute the function locally - 
https://github.com/hazelcast/hazelcast/pull/16636#discussion_r393635780

**Comment by webashutosh** [comment]
> @webashutosh PR looks great! One minor thing and I think we can merge. Thank you for the contribution again!

Thanks @mmedenjak! Pushed a commit for adding `@Nonnull` annotation
I would start working on `forEach()` implementation now

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by Holmistr** [comment]
In that case, after clean PR builder, we're good to merge the PR :)

**Comment by mmedenjak** [comment]
Thank you again for your contribution and feel free to continue with the next task of your choosing. `forEach` sounds good. I guess it generally would work even with the default implementation but adding optimisations for remote calls definitely makes sense. Make sure you document this as I guess some users might use the function to try and do some side-effects during iteration.

**Comment by webashutosh** [comment]
Oops! Not sure which test is failing as I don't have access to Jenkins

**Comment by mmedenjak** [comment]
Yes, it's an unrelated test failure, merging. Till next time!

**Pull request merged**